pyvista/core/validation/check.py

```diff
@@ -967,90 +890,48 @@ def check_is_iterable_of_some_type(
 
     See Also
     --------
-    check_is_instance
-    check_is_iterable
-    check_is_iterable_of_strings
+    check_instance
+    check_iterable
+    check_iterable_items
 
     Examples
     --------
     Check if a ``tuple`` only has ``int`` or ``float`` elements.
 
     >>> from pyvista.core import validation
-    >>> validation.check_is_iterable_of_some_type(
-    ...     (1, 2, 3.0), (int, float)
-    ... )
+    >>> validation.check_iterable_items((1, 2, 3.0), (int, float))
 
     Check if a ``list`` only has ``list`` elements.
 
     >>> from pyvista.core import validation
-    >>> validation.check_is_iterable_of_some_type([[1], [2], [3]], list)
+    >>> validation.check_iterable_items([[1], [2], [3]], list)
 
     """
-    check_is_iterable(iterable_obj, name=name)
+    check_iterable(iterable_obj, name=name)
     try:
-        [
-            check_is_instance(
-                item, some_type, allow_subclass=allow_subclass, name=f"All items of {name}"
+        any(
+            check_instance(
+                item, item_type, allow_subclass=allow_subclass, name=f"All items of {name}"
             )
             for item in iterable_obj
-        ]
-    except TypeError:
-        raise
-
-
-def check_is_iterable_of_strings(iterable_obj, /, *, allow_subclass=True, name='String Iterable'):
-    """Check if an iterable's items are all strings.
-
-    Parameters
-    ----------
-    iterable_obj : Iterable
-        Iterable to check.
-
-    allow_subclass : bool, default: True
-        If ``True``, the type of the iterable's items must be any of the
-        given types or a subclass thereof. Otherwise, subclasses are not
-        allowed.
-
-    name : str, default: "String Iterable"
-        Variable name to use in the error messages if any are raised.
-
-    Raises
-    ------
-    TypeError
-        If any of the elements have an incorrect type.
-
-    See Also
-    --------
-    check_is_iterable
-    check_is_string
-    check_is_string_in_iterable
-
-    Examples
-    --------
-    Check if a ``tuple`` only has ``str`` elements.
-
-    >>> from pyvista.core import validation
-    >>> validation.check_is_iterable_of_strings(("cat", "dog"))
+        )
 
-    """
-    try:
-        check_is_iterable_of_some_type(iterable_obj, str, allow_subclass=allow_subclass, name=name)
     except TypeError:
         raise
 
 
-def check_is_string_in_iterable(string_in, /, string_iterable, *, name='String'):
-    """Check if a given string is in an iterable of strings.
+def check_contains(item, /, container, *, name='Input'):
```

Comment:
Should the order of `item` and `container` be swapped here? The current order is consisntent with `item in containter` syntax, but every other `check` or `validate` function has the "object being tested" as the first arg. So maybe `containter` should be the first arg here?

Reply:
no strong opinion but if the signature was `def check_contains(*, item, container,  name='String`) the caller would be forced to explicit about which arg is which